### DISCUSSION TITLE
[Android] Is there a way to use a different toolchain for building packages and building the project?

### DISCUSSION THREAD

**Discussion by bmanga** [microsoft/vcpkg#36107] [Q&A]
Currently a lot of packages are broken using the latest ndk (r26b), but I need that version to build my project code.
Is there a way when invoking cmake to tell vcpkg to use the older ndk (25.xx) but use the newer one to build my project?

(Pinging @dg0yt as you seem to have android experience, sorry to bother)

**Comment by dg0yt** [comment]
I would not mix NDKs. Do they promise ABI compatibility for the C++ lib?
You should really try to use the prefixed compilers from NDK r26. I assume CMake won't complain, and autotools might be happy.

**Comment by bmanga** [comment]
Thanks for the reply :)
My understanding is that libc++ should commit to ABI stability. I managed to get my scenario to work by overriding the CMAKE_CXX_COMPILER and CMAKE_SYSROOT variables to the new ndk. Everything seems to be working fine.

I don't like this solution and it's just an interim hack until vcpkg supports ndk r26. 

> You should really try to use the prefixed compilers from NDK r26. I assume CMake won't complain, and autotools might be happy.

Could you explain how I can use the prefixed compilers? Just overriding the CMAKE_CXX_COMPILER ?
